src/runtime/policy/tests/policy-test.ts

```diff
@@ -0,0 +1,265 @@
+/**
+ * @license
+ * Copyright (c) 2020 Google Inc. All rights reserved.
+ * This code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt
+ * Code distributed by Google as part of this project is also
+ * subject to an additional IP rights grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+import {Manifest} from '../../manifest.js';
+import {assert} from '../../../platform/chai-web.js';
+import {PolicyEgressType, PolicyRetentionMedium, PolicyAllowedUsageType} from '../policy.js';
+import {assertThrowsAsync} from '../../../testing/test-util.js';
+import {mapToDictionary} from '../../util.js';
+
+// TODO(b/157605585): Test what happens with >1 description, egressType,
+// allowedRetention, allowedUsage.
+
+const customAnnotation = `
+annotation custom
+  targets: [Policy, PolicyTarget, PolicyField]
+  retention: Source
+  doc: 'custom annotation for testing'`;
+
+async function parsePolicy(str: string) {
+  const manifest = await Manifest.parse(customAnnotation + str);
+  assert.lengthOf(manifest.policies, 1);
+  return manifest.policies[0];
+}
+
+describe('policy', () => {
+  it('can round-trip to string', async () => {
+    const manifestString = `${customAnnotation}
+@intendedPurpose(description: 'test')
+@egressType(type: 'Logging')
+@custom
+policy MyPolicy {
+  @maxAge(age: '2days')
+  @allowedRetention(medium: 'Ram', encryption: false)
+  @allowedRetention(medium: 'Disk', encryption: true)
+  @custom
+  from SomeType access {
+    @allowedUsage(label: 'raw', usageType: 'join')
+    @allowedUsage(label: 'truncated', usageType: 'egress')
+    sensitiveField {
+      @allowedUsage(label: 'redacted', usageType: '*')
+      @custom
+      someField,
+    }
+    @custom
+    anotherField,
+  }
+  config SomeConfig {
+    abc: '123'
+    def: '456'
+  }
+}`.trim();
+    const manifest = await Manifest.parse(manifestString);
+    assert.strictEqual(manifest.toString(), manifestString);
+  });
+
+  it('rejects duplicate policy names', async () => {
+    await assertThrowsAsync(async () => parsePolicy(`
+policy MyPolicy {}
+policy MyPolicy {}
+`), 'A policy named MyPolicy already exists.');
+  });
+
+  it('policy annotations work', async () => {
+    const policy = await parsePolicy(`
+@intendedPurpose('test')
+@egressType('Logging')
+@custom
+policy MyPolicy {}
+`);
+    assert.strictEqual(policy.name, 'MyPolicy');
+    assert.strictEqual(policy.description, 'test');
+    assert.strictEqual(policy.egressType, PolicyEgressType.Logging);
+    assert.lengthOf(policy.annotations, 1);
+    assert.strictEqual(policy.annotations[0].name, 'custom');
+  });
+
+  it('rejects unknown egress types', async () => {
+    await assertThrowsAsync(async () => parsePolicy(`
+@egressType('SomethingElse')
+policy MyPolicy {}
+`), 'Expected one of: Logging, FederatedAggregation. Found: SomethingElse.');
```

Comment:
nit - maybe error message should say smth like "egressType expected one of..."

Reply:
It was tricky getting that to work with the `checkValueInEnum()` function, so not done